### PR TITLE
Replace erlang-sqlite3 with esqlite

### PR DESCRIPTION
**Pull request by maxmellen**

### CODE REVIEW

File: apps/arweave/src/ar_sqlite3.erl [hunk 1]
```diff
@@ -197,15 +196,11 @@ handle_call({eval_legacy_arql, Query}, _, State) ->
...
 				" ORDER BY block.height DESC, tx.id DESC"
 			]),
-			case sqlite3:prepare(?DB_NAME, SQL) of
-				{ok, Stmt} ->
-					ok = sqlite3:bind(?DB_NAME, Stmt, Params),
-					R = collect_txids(Stmt),
-					ok = sqlite3:finalize(?DB_NAME, Stmt),
-					R;
-				{error, 1, "parser stack overflow"} ->
-					sqlite_parser_stack_overflow;
```
Comment (ldmberman) [inline apps/arweave/src/ar_sqlite3.erl]: Could you, please, also update `ar_http_iface_middleware.erl` - it has some special handling for `sqlite_parser_stack_overflow`.

File: apps/arweave/src/ar_sqlite3.erl [hunk 2]
```diff
@@ -218,117 +213,100 @@ handle_cast({populate_db, BHL}, State) ->
...
 				maps:get(<<"height">>, BlockMap),
 				maps:get(<<"timestamp">>, BlockMap)
 			]),
-			done = sqlite3:next(?DB_NAME, Stmt),
+			ok = esqlite3:fetchone(Stmt),
 			ok = lists:foreach(fun(TXHash) ->
 				ok = insert_tx_json(TXHash, maps:get(<<"indep_hash">>, BlockMap), Env)
 			end, maps:get(<<"txs">>, BlockMap)),
-			ok = sqlite3:reset(?DB_NAME, Stmt),
+%%			ok = sqlite3:reset(?DB_NAME, Stmt),
```
Comment (ldmberman) [inline apps/arweave/src/ar_sqlite3.erl]: %%

File: rebar.config [hunk 3]
```diff
@@ -3,6 +3,7 @@
 	{jiffy, "0.15.2"},
 	{fusco, "0.1.0"},
 	{cowboy, "2.7.0"},
+	{esqlite, "0.4.0"},
 	{sqlite3, "1.1.6"},
```
Comment (ldmberman) [inline rebar.config]: We do not need to specify `sqlite3` here anymore I think?

File: apps/arweave/src/ar_sqlite3.erl [hunk 4]
```diff
@@ -218,117 +213,100 @@ handle_cast({populate_db, BHL}, State) ->
...
-		select_tags_by_tx_id_stmt := SelectTagsByTxIdStmt
-	} = State,
-	ok = sqlite3:finalize(?DB_NAME, SelectTagsByTxIdStmt),
-	ok = sqlite3:finalize(?DB_NAME, SelectBlockByTxIdStmt),
-	ok = sqlite3:finalize(?DB_NAME, SelectTxByIdStmt),
-	ok = sqlite3:finalize(?DB_NAME, InsertTagStmt),
-	ok = sqlite3:finalize(?DB_NAME, InsertTxStmt),
-	ok = sqlite3:finalize(?DB_NAME, InsertBlockStmt),
-	ok = sqlite3:close(?DB_NAME).
+	ok = sqlite3:close(Conn).
```
Comment (ldmberman) [inline apps/arweave/src/ar_sqlite3.erl]: Should it be `esqlite3:close`?

Comment (maxmellen) [reply apps/arweave/src/ar_sqlite3.erl]: Good catch 😅 f8648336

File: rebar.config (hunk 3 above)
Comment (maxmellen) [reply rebar.config]: Yep, forgot to remove it 👍 3c87e86

File: apps/arweave/src/ar_sqlite3.erl (hunk 2 above)
Comment (maxmellen) [reply apps/arweave/src/ar_sqlite3.erl]: 👍 e79ec3b9

File: apps/arweave/src/ar_sqlite3.erl (hunk 1 above)
Comment (maxmellen) [reply apps/arweave/src/ar_sqlite3.erl]: Added back 👍 ff3cf644

**Pull request merged**